addons/pos_paytm/static/src/js/payment_paytm.js

```diff
@@ -0,0 +1,146 @@
+odoo.define('pos_paytm.payment', function (require) {
+"use strict";
+
+const core = require('web.core');
+const rpc = require('web.rpc');
+const PaymentInterface = require('point_of_sale.PaymentInterface');
+const { Gui } = require('point_of_sale.Gui');
+
+const _t = core._t;
+
+let PaymentPaytm = PaymentInterface.extend({
+    init: function (pos, payment_method) {
+        this._super(...arguments);
+    },
+
+    send_payment_request: async function (cid) {
+        /**
+         * Override
+         */
+        this._super.apply(this, arguments);
+        const line = this.pos.get_order().selected_paymentline;
+        const order = this.pos.selectedOrder;
+        let retry = localStorage.getItem(order.uid);
+        if (!retry) {
+            retry = 0;
+            localStorage.setItem(order.uid, retry);
+        }
+        let orderId = order.name.replace(' ','').replaceAll('-','').toUpperCase();
+        if (retry > 0) {
+            orderId = orderId + 'retry' + retry;
+        }
+        const transactionAmount = line.amount*100;
+        const timeStamp = Math.floor(Date.now() / 1000);
+        const response = await this.makePaymentRequest(transactionAmount, orderId, timeStamp);
+        if (!response || response['body']['resultInfo']['resultCode'] === "F") {
+            let errorMessage = "Unable to establish connection with PayTM";
+            if (response) {
+                errorMessage = response['body']['resultInfo']['resultMsg'];
+            }
+            this._showError(errorMessage);
+            line.set_payment_status('force_done');
+            this._incrementRetry(order.uid);
+            throw false;
+        }
+        line.set_payment_status('waitingCard');
+        const pollresponse = await this.pollPayment(orderId, timeStamp);
+        if (pollresponse) {
+            localStorage.removeItem(order.uid);
+            return true;
+        }
+        else {
+            this._incrementRetry(order.uid);
+            return false;
+        }
+    },
+
+    send_payment_cancel: async function (order, cid) {
+        this._super.apply(this, arguments);
+        const line = this.pos.get_order().selected_paymentline;
+        line.set_payment_status('retry');
+        this._incrementRetry(order.uid);
+        clearTimeout(this.pollTimeout);
+        return true;
+    },
+
+
+    pollPayment: async function (transaction, timestamp) {
```

Comment:
A more efficient way in building a new module would be to use websocket to retrieve the status of your transaction.
so you don't have the disadvantages of pooling and the advantage of having the response in real time without waiting a maximum of 5 seconds

Reply:
We don't have any call back url OR  web-hook still websocket is good to have it?

Reply:
oh ? no webhook.
Ok 
and yes it is always interesting in terms of the execution time. Faster than waiting for 5 seconds, simpler too and more efficient in terms of performance.
Because websockets do not monopolize workers